google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -40,4 +83,206 @@ public String getCloudSdkHomePath() {
+
+  @NotNull
+  @Override
+  // TODO this path is incorrect. We need to determine an alternate strategy for loading these libs
+  // maybe from maven central
+  public VirtualFile[] getOrmLibSources() {
+    List<VirtualFile> roots = new ArrayList<>();
+    if (getJavaToolsBasePath() != null) {
+      final File libsDir = getJavaToolsBasePath().resolve(Paths.get("src", "orm")).toFile();
+      final File[] files = libsDir.listFiles();
+      if (files != null) {
+        for (File file : files) {
+          final String url = VfsUtil.getUrlForLibraryRoot(file);
+          final VirtualFile zipRoot = VirtualFileManager.getInstance().findFileByUrl(url);
+          if (zipRoot != null && zipRoot.isDirectory()) {
+            String fileName = file.getName();
+            final String srcDirName = StringUtil.trimEnd(fileName, "-src.zip");
+            final VirtualFile sourcesDir = zipRoot.findFileByRelativePath(srcDirName + "/src/java");
+            if (sourcesDir != null) {
+              roots.add(sourcesDir);
+            } else {
+              roots.add(zipRoot);
+            }
+          }
+        }
+      }
+    }
+    return VfsUtilCore.toVirtualFileArray(roots);
+  }
+
+  @Nullable
+  @Override
+  public File getWebSchemeFile() {
+    return getJavaToolsBasePath() != null
+        ? getJavaToolsBasePath().resolve(Paths.get("docs", "appengine-web.xsd")).toFile()
+        : null;
+  }
+
+  @Override
+  public void patchJavaParametersForDevServer(@NotNull ParametersList vmParameters) {
+    if (getJavaToolsBasePath() != null) {
+      File agentPath = getJavaToolsBasePath().resolve(
+          Paths.get("lib", "agent", "appengine-agent.jar")).toFile();
+      if (agentPath.exists()) {
+        vmParameters.add("-javaagent:" + agentPath.getAbsolutePath());
+      }
+      File patchPath = getJavaToolsBasePath().resolve(
+          Paths.get("lib", "override", "appengine-dev-jdk-overrides.jar")).toFile();
+      if (patchPath.exists()) {
+        vmParameters.add("-Xbootclasspath/p:" + patchPath.getAbsolutePath());
+      }
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getVersion() {
+    return getToolsApiJarFile() != null
+        ? JarUtil.getJarAttribute(
+            getToolsApiJarFile(),
+            "com/google/appengine/tools/info/", // The trailing slash is needed so don't use Path
+            Attributes.Name.SPECIFICATION_VERSION)
+        : null;
+  }
+
+  @Override
+  public boolean isValid() {
+    return !StringUtil.isEmpty(getCloudSdkHomePath())
+        && getToolsApiJarFile() != null
+        && getToolsApiJarFile().exists();
+  }
+
+  private static String findLatestVersion(File dir) {
+    String[] names = dir.list();
+    if (names != null && names.length > 0) {
+      String max = Collections.max(Arrays.asList(names));
+      return FileUtil.toSystemIndependentName(new File(dir, max).getAbsolutePath());
+    }
+    return null;
+  }
+
+  private Map<String, Set<String>> loadBlackList() throws IOException {
```

Comment:
wait wasn't this replaced by the whitelist?

On Thu, Aug 18, 2016 at 3:10 PM, Etan Shaul <notifications@github.com>
wrote:

> In google-cloud-tools-plugin/src/com/google/cloud/tools/
> intellij/appengine/sdk/DefaultCloudSdkService.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/822#discussion_r75370270>
> :
>
> > +  public boolean isValid() {
> > +    return !StringUtil.isEmpty(getCloudSdkHomePath())
> > +        && getToolsApiJarFile() != null
> > +        && getToolsApiJarFile().exists();
> > +  }
> > +
> > +  private static String findLatestVersion(File dir) {
> > +    String[] names = dir.list();
> > +    if (names != null && names.length > 0) {
> > +      String max = Collections.max(Arrays.asList(names));
> > +      return FileUtil.toSystemIndependentName(new File(dir, max).getAbsolutePath());
> > +    }
> > +    return null;
> > +  }
> > +
> > +  private Map<String, Set<String>> loadBlackList() throws IOException {
>
> do you know where this method blacklist even comes from? It was pulled
> from old plugin, but I don't see this documented anywhere in AE (whereas
> the class whitelist is documented).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/822/files/0bbf4e19de562c3c734710e081ac539e182b4686#r75370270>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HVnuJ9nOTU704gdr0w1Oj5dee16dks5qhK4WgaJpZM4Jlk8z>
> .
>